framework/src/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -537,14 +539,26 @@ public Boolean visitTypeArgs(
                 return true;
             }
 
-            if (supertypeTypeArgs.size() > 0) {
-                for (int i = 0; i < supertypeTypeArgs.size(); i++) {
-                    final AnnotatedTypeMirror superTypeArg = supertypeTypeArgs.get(i);
-                    final AnnotatedTypeMirror subTypeArg = subtypeTypeArgs.get(i);
-                    if (!compareTypeArgs(
-                            subTypeArg, superTypeArg, supertypeRaw, subtypeRaw, visited)) {
-                        return false;
-                    }
+            final TypeElement supertypeElem =
+                    (TypeElement) supertype.getUnderlyingType().asElement();
+            List<Integer> covariantArgIndexes = null;
+            AnnotationMirror covam =
+                    supertype.atypeFactory.getDeclAnnotation(supertypeElem, Covariant.class);
+            if (covam != null) {
+                covariantArgIndexes =
+                        AnnotationUtils.getElementValueArray(covam, "value", Integer.class, false);
+            }
+
+            for (int i = 0; i < supertypeTypeArgs.size(); i++) {
+                final AnnotatedTypeMirror superTypeArg = supertypeTypeArgs.get(i);
+                final AnnotatedTypeMirror subTypeArg = subtypeTypeArgs.get(i);
+                final boolean covariant =
+                        this.covariantTypeArgs
```

Comment:
Thanks, I removed the field. It seems much nicer to override that method.
I also added the old annotation back again but deprecated it, to make the transition easier.

All tests pass locally for me and the Travis error doesn't really make sense.
@smillst do you see what is wrong on Travis? Only thing I can think of is downloading the JDK, but that also doesn't make sense, in particular now with the deprecated annotation.